src/context.rs

```diff
@@ -422,7 +422,7 @@ impl<A: Actor> Context<A> {
                 let delay = Delay::new(duration);
                 match future::select(delay, &mut stopped).await {
                     Either::Left(_) => {
-                        if addr.do_send(constructor()).is_err() {
+                        if addr.send(constructor()).await.is_err() {
```

Comment:
This (as well as the change on L458) would change how the methods work w.r.t backpressure. Whilst the old behaviour was not explicitly documented, it might be worth noting in the doc comment now that it will be changing from how it used to work.

Reply:
This, however, would be irrelevant if split out into an extension crate